src/attestation/Attestation.chain.ts

```diff
@@ -29,23 +29,31 @@ export async function store(
 
   const blockchain = await getCached()
 
-  // @ts-ignore
-  const tx: SubmittableExtrinsic<
-    CodecResult,
-    any
-  > = await blockchain.api.tx.attestation.add(
+  const tx = await blockchain.api.tx.attestation.add(
     txParams.claimHash,
     txParams.ctypeHash,
     txParams.delegationId
   )
   return blockchain.submitTx(identity, tx)
 }
 
-export async function query(
+function decode(
+  encoded: QueryResult,
   claimHash: string
-): Promise<Attestation | undefined> {
-  const encoded: QueryResult = await queryRaw(claimHash)
-  return decode(encoded, claimHash)
+): Attestation | undefined {
+  if (encoded && encoded.encodedLength) {
+    const attestationTuple = encoded.toJSON()
+    const attestation: IAttestation = {
+      claimHash,
+      cTypeHash: attestationTuple[0],
+      owner: attestationTuple[1],
+      delegationId: attestationTuple[2],
+      revoked: attestationTuple[3],
+    }
+    log.info(`Decoded attestation: ${JSON.stringify(attestation)}`)
+    return Attestation.fromObject(attestation)
+  }
+  return undefined
```

Comment:
Nothing that's in the scope of this PR, but:    
General question: why are we using `undefined` over `null` in the SDK codebase?  
 
So far I was told that `undefined` should rather be left to the compiler to signify actual `undefined` errors, and use `null` instead.

Reply:
yes, we should switch to null

Reply:
https://github.com/KILTprotocol/sdk-js/issues/153